openapi.yml

```diff
@@ -32,71 +55,85 @@ paths:
                 type: array
                 items:
                   $ref: "#/components/schemas/OrderFull"
-  /api/v1/nonce:
+  /api/v1/gas-tip/{sellToken}:
     get:
-      summary: The nonce to pick for a new order.
-      description: The best nonce to pick depends on how much time there is left in the current settlement according to the operator. This time is token-pair specific which is why the sell and buy tokens must be supplied.
-      parameters:
-        - name: sellToken
-          in: query
-          schema:
-            $ref: "#/components/schemas/Address"
-        - name: buyToken
-          in: query
-          schema:
-            $ref: "#/components/schemas/Address"
+      summary: The tip that is used to pay solvers for their service in the sellToken of an order.
+      description: The tip is calculated by the order service and depends on the current gas price and the expected gas costs to settle the trade
       responses:
         200:
-          description: the nonce
+          description: the tip
           content:
             application/json:
               schema:
-                $ref: "#/components/schemas/Nonce"
+                $ref: "#/components/schemas/Tip"
+        400:
+          description: sellToken not existent
 components:
   schemas:
     Address:
       description: Ethereum 40 byte address encoded as a hex without `0x` prefix.
       type: string
       example: "6810e776880c02933d47db1b9fc05908e5386b96"
     TokenAmount:
-      description: Amount of a token. uint112 encoded in decimal.
+      description: Amount of a token. uint256 encoded in decimal.
       type: string
       example: "1234567890"
-    Nonce:
-      description: Nonce of the batch in which the order is valid. uint32.
-      type: integer
-    OrderNew:
+    Tip:
+      description: The tip that is used to pay solvers. uint256.
+      type: object
+      properties:
+        expirationDate:
+          description: |
+            Expiration date of the offered tip. Order service might not accept 
+            the tip after this expiration date. Encoded as unix timestamp.
+          type: integer
+        tipAmount:
+          description: Amount of tip in specified sellToken
+          $ref: "#/components/schemas/TokenAmount"
+    OrderType:
+      description: Is this a buy order or sell order?
+      type: string
+      enum: [buy, sell]
+    OrderCreation:
```

Comment:
There's no `owner` in `OrderCreation`. Why and can we add it?

Reply:
The owner is implicitly provided by (and extracted from) the signature. It is part of OrderFull, which is the struct that should be used for querying information.

Reply:
Signature is good to have,but it would be really nice to be able to immediately recognize and owner from the Order. Unless the signature itself is `<owner>_<uuid>` or something

Reply:
Since `OrderCreation` is just used as POST data for adding a new orders, I'm not sure I understand the value in requiring API clients to also post the `owner` if they already have to compute a ECDSA signature with a wallet that surely can also compute its public address.

> Signature is good to have,but it would be really nice to be able to immediately recognize the owner from the Order.

As @fleupold mentioned queried orders do contain an `owner` field representing the public address of the creator of the order. It is computed by recovering the public key of the private key used to sompute the ECDSA signature :point_up: 

> Unless the signature itself is <owner>_<uuid> or something

I think this is the ECDSA signature, so 65 bytes (`v || r || s`).

Reply:
> the value in requiring API clients to also post the owner
> queried orders do contain an owner field

Not for posting to the endpoint, but more for querying for all orders from all owners, to later be able to filter by owner address or some such

Reply:
> but more for querying for all orders from all owners

When querying, you will always get an instance of `OrderFull` which contains owner information, creation timestamp and other things that the `OrderCreation` payload doesn't expose.

Reply:
In that case, it should already be there. The type of a queried order is `OrderFull` which has `owner: Address` field:

https://github.com/gnosis/oba-services/blob/deacab85a1e9f797ada0e579e3f2fb82e8953a6d/openapi.yml#L133-L144

Reply:
Cool, somehow I got confused which `Order*` is returned where